File: mapscript/python/examples/shpdump.py
#!/usr/bin/env python

""" Dump the contents of the passed in shapefile.

    Pass in the basename of the shapefile.  This script uses the mapscript
    extention module.
"""

import mapscript
import sys
import os

# Utility functions.
def usage():
  """ Display usage if program is used incorrectly. """
  print "Syntax: %s base_filename" % sys.argv[0]
  sys.exit(2)

def plural(x):
  """ Returns an 's' if plural. 

  Useful in print statements to avoid something like 'point(s)'.  """
  if x > 1:
    return 's'
  return ''


# Make sure passing in filename argument.
if len(sys.argv) != 2:
  usage()

# Make sure can access .shp file, create shapefileObj.
if os.access(sys.argv[1] + ".shp", os.F_OK):
  sf_obj = mapscript.shapefileObj(sys.argv[1], -1)
else:
  print "Can't access shapefile"
  sys.exit(2)

# Create blank shape object.
s_obj = mapscript.shapeObj(-1)

# Loop through each shape in the shapefile.
for i in range(sf_obj.numshapes):

  # Get the ith object.
  sf_obj.get(i, s_obj)
  print "Shape %i has %i part%s." % (i, s_obj.numlines, plural(s_obj.numlines))
  print "bounds (%f, %f) (%f, %f)" % (s_obj.bounds.minx, s_obj.bounds.miny, s_obj.bounds.maxx, s_obj.bounds.maxy)

  # Loop through parts of each shape.
  for j in range(s_obj.numlines):

    # Get the jth part of the ith object.
    part = s_obj.get(j)
    print "Part %i has %i point%s." % (j, part.numpoints, plural(part.numpoints))

    # Loop through points in each part.
    for k in range(part.numpoints):

      # Get the kth point of the jth part of the ith shape.
      point = part.get(k)
      print "%i: %f, %f" % (k, point.x, point.y)

File: mapscript/csharp/examples/shpdump.cs
using System;

/**
 * <p>Title: Mapscript shape dump example.</p>
 * <p>Description: A Java based mapscript to dump information from a shapefile.</p>
 * @author Yew K Choo  (ykchoo@geozervice.com)
 * @version 1.0
 */

/// <summary>
/// A C# based mapscript mapscript to dump information from a shapefile.
/// </summary>  
class ShapeDump {
  public static void usage() 
  { 
	Console.WriteLine("usage: shpdump {shapefile}");
	System.Environment.Exit(-1);
  }
  	
  public static void Main(string[] args) {

	if (args.Length != 1) usage();
		  
 	shapefileObj sf_obj = new shapefileObj(args[0],-1); 
	shapeObj s_obj = new shapeObj(-1);
	for (int i=0; i<sf_obj.numshapes; i++) 
	{
	  sf_obj.get(i, s_obj);
	  Console.WriteLine("Shape " + i + " has " + s_obj.numlines + " part(s)");
	  Console.WriteLine("bounds (" + s_obj.bounds.minx + ", " +  s_obj.bounds.miny + ") (" + s_obj.bounds.maxx + ", " + s_obj.bounds.maxy + ")" );
	  for(int j=0; j<s_obj.numlines; j++) 
	  {
		lineObj l_obj = s_obj.get(i);
		Console.WriteLine("Part " + j + " has " + l_obj.numpoints + " points.");
                    
		for(int k=0; k < l_obj.numpoints; k++) {
		  pointObj p_obj = l_obj.get(k);                     
          Console.WriteLine(k +": " + p_obj.x + ", " + p_obj.y);							
		}
	  }	  
    }
  }
}